### PR TITLE
Add second custom color option

### PR DESCRIPTION
**Pull request by laurelfulford**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The secondary colour has only been added to the content links so far (which alone, needed a lot of variables to be changed). We still need to finalize exactly where it should be applied otherwise in the theme, and added there in a future PR. 

This PR also:

* Cleans up comments in color-patterns.php, to reference the colour being applied rather than what it's being applied to. 
* Removes theme's Dark Grey and Light Grey from the editor colour palette; we will want to have some default colours that are always there, but we'll need to figure out what they should be once the default colour palettes are finalized. As is, with a default secondary grey colour, you end up with a lot of different greys. 
* Removes the `postMessage` transport from the Customizer colour options, and the JavaScript preview. Unfortunately, incorporating multiple colours makes previewing like this more complex -- it's something to revisit in the future, once the colour behaviour is finalized, but is not a breaking issue (the Customizer will just do a full refresh when the colour is changed, rather than previewing without). 

See #24.

### How to test the changes in this Pull Request:

1. Apply the PR.
2. Confirm that there is a mid-range and dark grey added to the editor as colour options.
3. In the Customizer, confirm that there is an option for a second custom colour. 
4. Change the colour; confirm it changes links in the content body, and the colour options in the editor. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -154,14 +155,14 @@ function newspack_setup() {
 					'color' => 'default' === get_theme_mod( 'theme_colors' ) ? newspack_adjust_brightness( $primary_color, -40 ) : newspack_adjust_brightness( get_theme_mod( 'primary_color_hex', $primary_color ), -40 ),
 				),
 				array(
-					'name'  => __( 'Dark Gray', 'newspack' ),
-					'slug'  => 'dark-gray',
-					'color' => '#111',
+					'name'  => __( 'Secondary', 'newspack' ),
```
Comment (philipjohn) [inline functions.php]: Minor: It's always safer to use `esc_html__()` and adjust later if required for translations with special characters.

File: functions.php [hunk 2]
```diff
@@ -154,14 +155,14 @@ function newspack_setup() {
...
+					'slug'  => 'secondary',
+					'color' => 'default' === get_theme_mod( 'theme_colors' ) ? $secondary_color : get_theme_mod( 'secondary_color_hex', $secondary_color ),
 				),
 				array(
-					'name'  => __( 'Light Gray', 'newspack' ),
-					'slug'  => 'light-gray',
-					'color' => '#767676',
+					'name'  => __( 'Secondary Variation', 'newspack' ),
```
Comment (philipjohn) [inline functions.php]: Minor: It's always safer to use `esc_html__()` and adjust later if required for translations with special characters.

File: functions.php [hunk 3]
```diff
@@ -154,14 +155,14 @@ function newspack_setup() {
 					'color' => 'default' === get_theme_mod( 'theme_colors' ) ? newspack_adjust_brightness( $primary_color, -40 ) : newspack_adjust_brightness( get_theme_mod( 'primary_color_hex', $primary_color ), -40 ),
 				),
 				array(
-					'name'  => __( 'Dark Gray', 'newspack' ),
-					'slug'  => 'dark-gray',
-					'color' => '#111',
+					'name'  => __( 'Secondary', 'newspack' ),
+					'slug'  => 'secondary',
+					'color' => 'default' === get_theme_mod( 'theme_colors' ) ? $secondary_color : get_theme_mod( 'secondary_color_hex', $secondary_color ),
```
Comment (philipjohn) [inline functions.php]: Nitpick: It'd be more readable to have this broken up across multiple lines.

File: inc/color-filters.php [hunk 4]
```diff
@@ -21,10 +22,30 @@ function newspack_get_primary_color() {
 }
 
 /**
- * Tests the current default hue against NEWSPACK_DEFAULT_HEX.
+ * The default color used for the secondary color throughout this theme
+ *
+ * @return string the default hexidecimal color.
+ */
+function newspack_get_secondary_color() {
+	return apply_filters( 'newspack_secondary_color', NEWSPACK_DEFAULT_SECONDARY );
```
Comment (philipjohn) [inline inc/color-filters.php]: We should document this hook according to the [inline docs standards](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters).

File: functions.php (hunk 1 above)
Comment (laurelfulford) [reply functions.php]: Ah, good call! This is throughout the theme -- strings marked for translation aren't escaped in default themes, but it is standard practice on WP.com. I've created a separate issue to tackle this (#30), and am going to mark this comment as resolved.

File: functions.php (hunk 2 above)
Comment (laurelfulford) [reply functions.php]: Same as the previous comment -- let's fix this up throughout in #30!

File: functions.php (hunk 3 above)
Comment (laurelfulford) [reply functions.php]: Good call! Tidied up in f484753.

File: inc/color-filters.php (hunk 4 above)
Comment (laurelfulford) [reply inc/color-filters.php]: Good catch - I've added comment for both this new colour filter and the one that was already there.

**Pull request merged**